### PR TITLE
1D Plots

### PR DESCRIPTION
**Pull request by corranwebster**
This is the start of a general infrastructure for plots which only distribute their data in one dimension.

The motivation are things like plotting tops and associated descriptions in geophysics files, but also allowing things like axis scatters as used by Tufte.  Potentially can move jitter plots and colorbar plots into this infrastructure.

This is not yet ready to merge.

### CODE REVIEW

File: chaco/jitterplot.py [hunk 1]
```diff
@@ -9,98 +9,27 @@
...
     """A renderer for a jitter plot, a 1D plot with some width in the
     dimension perpendicular to the primary axis.  Useful for understanding
     dense collections of points.
     """
 
-    # The data source of values
-    index = Instance(ArrayDataSource)
-
-    # The single mapper that this plot uses
-    mapper = Instance(AbstractMapper)
```
Comment (corranwebster) [inline chaco/jitterplot.py]: Fixed that, I think (I replaced it with an appropriate use of `RandomState`).

**Comment by corranwebster** [comment]
Basic smoke-tests in place.

**Comment by cfarrow** [comment]
It look like all comments are addressed. I'd like to start using this and propose this be merged. @jonathanrocher any objections?

**Comment by jonathanrocher** [comment]
As said offline, as long as the demo runs and travis is OK, I think this is ok to merge. Currently the demo fails to run on my machine with 
```
  File "/Users/jrocher/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/traitsui/editor.py", line 133, in __init__
    self.old_value = getattr( self.object, self.name )
  File "examples/demo/basic/scatter_1d.py", line 116, in _plot_default
    return _create_plot_component()
  File "examples/demo/basic/scatter_1d.py", line 40, in _create_plot_component
    plot.plot_1d(
AttributeError: 'Plot' object has no attribute 'plot_1d'
```
@cfarrow do you have the bandwidth to take a look at that and at the kiwisolver issue reported by Travis?

**Comment by cfarrow** [comment]
The `AttributeError` is due to not running `setup.py`, I believe. (I believe this because I forgot to run `setup.py`. ;))

I'll look into the kiwisolver issue.

**Comment by jonathanrocher** [comment]
Argh, thanks, the demo is running now. Sorry for the noise.

**Comment by jwiggins** [comment]
@cfarrow: kiwisolver doesn't appear to be in Chaco's `dev_requirements.txt`. I would lean on the Travis bootstrap for Enable if you haven't already done this. I seem to recall some recent hair pulling related to dependency resolution on Travis.

**Comment by cfarrow** [comment]
I'll take a look. Thanks @jwiggins!

**Comment by jwiggins** [comment]
Just to elaborate, the hair pulling is here: enthought/enable#178

**Comment by cfarrow** [comment]
@jwiggins I was able to add the `kiwisolver` dependency and get a clean test run. I did not investigate the TravisCI setup beyond that. That is an effort of its own, so I'm going to punt on that for now unless you strongly object.

**Comment by jwiggins** [comment]
Probably best to leave well enough alone in this PR

**Comment by cfarrow** [comment]
Agreed. :sweat_smile: 

Thanks.

**Pull request merged**